src/sync/incomingSyncUtils.js

```diff
@@ -557,13 +557,16 @@ export const createOrUpdateRecord = (database, settings, recordType, record) =>
       try {
         const parsedData = JSON.parse(json);
         const shouldSetData = checkIsObject(parsedData);
-        internalRecord = {
-          id,
-          title,
-          type,
-          _data: shouldSetData ? JSON.stringify(parsedData.data) : null,
-        };
-        database.update(recordType, internalRecord);
+        const validatedReport = validateReport(parsedData, type);
```

Comment:
does the `validatedReport` now remove the need for `shouldSetData`?

Is it also not even needed now to parse the data? Maybe we could just do that in the `validateReport` helper?

Reply:
I was thinking about something like this:

```
case 'Report': {
      const { ID: id, title, type, json } = record;
      try {
        const parsedData = JSON.parse(json);
        const shouldSetData = checkIsObject(parsedData) ? validateReport(parsedData, type) : false;
        if (shouldSetData) {
          internalRecord = {
            id,
            title,
            type,
            _data: JSON.stringify(parsedData.data)
          };
          database.update(recordType, internalRecord);
        }
      } catch (error) {
        // Throw to parent, for now
        throw error;
      }
      break;
    }
```

Reply:
Done here df1e5b9
Open to suggestions anyway 😁